### PR TITLE
Switch to renovate for dependency management

### PR DESCRIPTION
**Pull request by NotMyFault**
When filing PRs like https://github.com/jenkinsci/archetypes/pull/596, I wondered if dependabot was just slow, because it didn't pick the directory up at all.
Screenshot 2023-03-10 at 15 38 18
Figured, dependabot expects a pom.xml here. Hence, there have been no automatic PRs before.

This PR replaces dependabot with renovate, which is fairly capable to update dependencies outside the typical build management files and paths. Hence, the config proposed is way smaller than the old dependabot one.

I requested an installation of renovate on this repository.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -47,7 +47,7 @@
         <module>empty-plugin</module>
         <module>hello-world</module>
         <module>global-configuration</module>
-        <!-- if adding new archetypes please also update .github/dependabot.yml -->
+        <!-- if adding new archetypes please also update .github/renovate.json -->
```
Comment (timja) [inline pom.xml:50]: is this comment still needed? won't they just be auto-detected?

**Review by timja — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (NotMyFault) [reply pom.xml]: > won't they just be auto-detected?

Absolutely, this was introduced when refactoring the file through IJ.

**Review by basil — DISMISSED** [review]
Lacks a justification for why we would switch from a first-party GitHub platform component to a third-party tool.

Lacks a migration plan for migrating the entire Jenkins ecosystem to the proposed end state.

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
> Lacks a migration plan for migrating the entire Jenkins ecosystem to the proposed end state.

This isn't updating the archetypes itself only this repository. Archetypes still use dependabot

**Comment by basil** [comment]
Fine then, I do not object strongly but I am still weakly against this change.

**Pull request merged**